Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -192,6 +155,22 @@ void _CGPathGetBoundingBoxInternal(CGPathRef pathref, CGRect* rectOut) {
     pathref->_getBoundingBox(rectOut);
 }
 
+void _CGPathAddElement(
+    CGPathRef path, CGPathElementType type, CGPoint p0 = CGPointZero, CGPoint p1 = CGPointZero, CGPoint p2 = CGPointZero) {
+    if (path->_count + 1 >= path->_max) {
+        path->_max += 32;
+        path->_elements = (CGPathElement*)IwRealloc(path->_elements, path->_max * sizeof(CGPathElement));
+    }
+    CGPathElement* element = &path->_elements[path->_count];
+    element->points = (CGPoint*)IwCalloc(kCGPathMaxPointCount, sizeof(CGPoint));
```

Comment:
This is not what i meant.  We are still allocating memory every time a path element is added.  perhaps i misunderstood your earlier question.  Do you want to chat over the phone so we can get on the same page here?  I really want to avoid duplicate allocations for points every time an item is added to the path.  Allocating a fixed number of points without doing it once isn't really buying us anything.